packages/medusa/src/api-v2/admin/price-lists/queries/list-price-lists.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  MedusaContainer,
+  PriceListRuleDTO,
+  PriceSetMoneyAmountDTO,
+} from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+import { cleanResponseData } from "../../../../utils/clean-response-data"
+import { adminPriceListRemoteQueryFields } from "../query-config"
+import { AdminPriceListRemoteQueryDTO } from "../types"
+
+export async function listPriceLists({
+  container,
+  fields,
+  variables,
+}: {
+  container: MedusaContainer
+  fields: string[]
+  variables: Record<string, any>
+}): Promise<[AdminPriceListRemoteQueryDTO[], number]> {
+  const remoteQuery = container.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+  const queryObject = remoteQueryObjectFromString({
+    entryPoint: "price_list",
+    fields: adminPriceListRemoteQueryFields,
+    variables,
+  })
+
+  const { rows: priceLists, metadata } = await remoteQuery(queryObject)
+
+  if (!metadata.count) {
+    return [[], 0]
+  }
+
+  for (const priceList of priceLists) {
+    priceList.rules = buildPriceListRules(priceList.price_list_rules || [])
+    priceList.prices = buildPriceSetPrices(
+      priceList.price_set_money_amounts || []
+    )
+  }
+
+  const sanitizedPriceLists: AdminPriceListRemoteQueryDTO[] = priceLists.map(
+    (priceList) => cleanResponseData(priceList, fields)
+  )
+
+  return [sanitizedPriceLists, metadata.count]
+}
+
+function buildPriceListRules(
+  priceListRules: PriceListRuleDTO[]
+): Record<string, string[]> {
+  return priceListRules.reduce((acc, curr) => {
+    const ruleAttribute = curr.rule_type.rule_attribute
+    const ruleValues = curr.price_list_rule_values || []
+
+    if (ruleAttribute) {
+      acc[ruleAttribute] = ruleValues.map((ruleValue) => ruleValue.value)
+    }
+
+    return acc
+  }, {})
+}
+
+function buildPriceSetPrices(
+  priceSetMoneyAmounts: PriceSetMoneyAmountDTO[]
+): Record<string, any>[] {
+  return priceSetMoneyAmounts.map((priceSetMoneyAmount) => {
+    const productVariant = (priceSetMoneyAmount.price_set as any).variant
```

Comment:
Why do we need the cast to any here, and can it be avoided?

Reply:
This is a deep remote link, we don't have any readily available types for this yet. 😄  We also need to figure out how to handle remote query types. I've temporarily added a partial type on the function and removed the cast. Will come back to this in the later PRs.